pkg/build/api/types.go

```diff
@@ -70,6 +70,20 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+	// Optional field that holds information about a builds' trigger.
+	BuildTriggerInfo BuildTriggerInfo
+}
+
+// BuildTriggerInfo is an optional field that stores information abou a
+// triggered build.
+type BuildTriggerInfo struct {
+	User                    string
+	BuildTriggerType        string
+	BuildTriggerInformation string
```

Comment:
@mfojtik this is just a sketch, i just want to know why i can't pass the information from generator.go
i can't figure out if there's any request validation or anything else between the controllers and the generator that would render the struct empty